services/datamanager/datasync/sync.go

```diff
@@ -56,7 +57,7 @@ type ManagerConstructor func(logger golog.Logger, cfg *config.Config, lastModMil
 
 // NewDefaultManager returns the default Manager that syncs data to app.viam.com.
 func NewDefaultManager(logger golog.Logger, cfg *config.Config, lastModMillis int) (Manager, error) {
-	if cfg.Cloud == nil || cfg.Cloud.AppAddress == "" {
+	if cfg.Cloud == nil || cfg.Cloud.SignalingAddress == "" {
```

Comment:
What is the signaling address?

Reply:
The signaling address in the config is either "app.viam.com:port" or "app.viam.dev:port"

Reply:
But maybe @edaniels can explain the difference whether signaling address and app address should be used

Reply:
It seems like app address is often empty in the cloud config (ex: [here](https://app.viam.com/api/json1/config?id=4be27022-9d23-4fa5-8fa1-5d5fba16eefb&full=t&client=t) and [here](https://app.viam.com/api/json1/config?id=a6f8d7b1-8b39-4ac0-a52a-0808f970a80a&full=t&client=t))

Reply:
that seems like a bug

Reply:
I went back through the robots I've created and noticed this is true